warehouse/integrations/redshift/redshift_test.go

```diff
@@ -999,6 +999,45 @@ func TestIntegration(t *testing.T) {
 			)
 			require.Equal(t, records, whth.SampleTestRecords())
 		})
+
+		t.Run("crashRecover", func(t *testing.T) {
+			ctx := context.Background()
+			tableName := "crash_recovery_test_table"
+			stgTableName := warehouseutils.StagingTableName(warehouseutils.RS, tableName, 64)
+			mockUploader := newMockUploader(t, nil, tableName, schemaInUpload, schemaInUpload, warehouseutils.LoadFileTypeParquet)
+
+			rs := redshift.New(config.New(), logger.NOP, stats.NOP)
+			err := rs.Setup(ctx, warehouse, mockUploader)
+			require.NoError(t, err)
+
+			err = rs.CreateSchema(ctx)
+			require.NoError(t, err)
+
+			err = rs.CreateTable(ctx, stgTableName, schemaInWarehouse)
+			require.NoError(t, err)
+
+			tableExists := func(t *testing.T) bool {
+				t.Helper()
+				var count int
+				err = rs.DB.DB.QueryRow(`
+					SELECT count(*)
+					FROM information_schema.tables
+					WHERE table_schema = $1 AND table_name = $2;
+						`,
+					namespace,
+					stgTableName,
+				).Scan(&count)
+				require.NoError(t, err)
+
+				return count == 1
+			}
+			require.True(t, tableExists(t))
+
+			err = rs.CrashRecover(ctx)
```

Comment:
Should we run this? twice to see, if it working as expected, Basically to test for `IF EXISTS`?

Reply:
the thing is to test `IF EXISTS` you need to run multiple CrashRecover methods in parallel and time it correctly. The issue happens when the table is deleted between the execution of two queries and that is hard to test consistently and reliably.

So here I am just testing CrashRecover, but not the specific issue.